src/Symfony/Component/Translation/Extractor/PhpAstExtractor.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Extractor;
+
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitor;
+use PhpParser\Parser;
+use PhpParser\ParserFactory;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Translation\Extractor\Visitor\ConstraintVisitor;
+use Symfony\Component\Translation\Extractor\Visitor\TranslatableMessageVisitor;
+use Symfony\Component\Translation\Extractor\Visitor\TransMethodVisitor;
+use Symfony\Component\Translation\Extractor\Visitor\Visitor;
+use Symfony\Component\Translation\MessageCatalogue;
+
+/**
+ * PhpAstExtractor extracts translation messages from a PHP AST.
+ *
+ * @author Mathieu Santostefano <msantostefano@protonmail.com>
+ */
+class PhpAstExtractor extends AbstractFileExtractor implements ExtractorInterface
+{
+    private string $prefix = '';
+    private Parser $parser;
+    /**
+     * @var array<Visitor&NodeVisitor>
+     */
+    private array $visitors;
+
+    public function __construct()
+    {
+        $this->parser = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
```

Comment:
since the dep is optional, we should throw a logic exception when it's missing, see L76 for inspiration